From b742b07e84939ea09ef83d3df7b1ec18e3946d1a Mon Sep 17 00:00:00 2001
From: William Wu <william.wu@rock-chips.com>
Date: Thu, 25 Jan 2018 11:23:44 +0800
Subject: [PATCH] usb: dwc_otg_310: increase otg max_transfer_size to 128KB

The default value of the max_transfer_size is only 65535,
but according to the field of PktSizeWidth in GHWCFG3, the
otg controller on rockchip platforms can support max 512KB
transfer size. So we increase the otg max_transfer_size to
128KB, it can help to avoid dividing into two hardware
transmissions if the usb device class driver(e.g. f_mtp.c)
request to transfer more than 65535 Bytes data in a single
usb request.

Change-Id: Ic281965815e48e4a146a13f9d37f88a10ee0d33e
Signed-off-by: William Wu <william.wu@rock-chips.com>
---
 drivers/usb/dwc_otg_310/dwc_otg_driver.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/usb/dwc_otg_310/dwc_otg_driver.c b/drivers/usb/dwc_otg_310/dwc_otg_driver.c
index 76f08979854c..19a0a06abee4 100644
--- a/drivers/usb/dwc_otg_310/dwc_otg_driver.c
+++ b/drivers/usb/dwc_otg_310/dwc_otg_driver.c
@@ -169,7 +169,7 @@ static struct dwc_otg_driver_module_params dwc_otg_module_params = {
 	.host_rx_fifo_size = -1,
 	.host_nperio_tx_fifo_size = -1,
 	.host_perio_tx_fifo_size = -1,
-	.max_transfer_size = -1,
+	.max_transfer_size = 131072,
 	.max_packet_count = -1,
 	.host_channels = -1,
 	.dev_endpoints = -1,
-- 
2.35.3

